airflow/providers/google/cloud/transfers/s3_to_gcs.py

```diff
@@ -221,3 +246,104 @@ def gcs_to_s3_object(self, gcs_object: str) -> str:
         if self.apply_gcs_prefix:
             return self.prefix + s3_object
         return s3_object
+
+    def transfer_files(self, s3_objects: list[str], gcs_hook: GCSHook, s3_hook: S3Hook):
+        if s3_objects:
+            dest_gcs_bucket, dest_gcs_object_prefix = _parse_gcs_url(self.dest_gcs)
+            for obj in s3_objects:
+                # GCS hook builds its own in-memory file, so we have to create
+                # and pass the path
+                file_object = s3_hook.get_key(obj, self.bucket)
+                with NamedTemporaryFile(mode="wb", delete=True) as file:
+                    file_object.download_fileobj(file)
+                    file.flush()
+                    gcs_file = self.s3_to_gcs_object(s3_object=obj)
+                    gcs_hook.upload(dest_gcs_bucket, gcs_file, file.name, gzip=self.gzip)
+
+            self.log.info("All done, uploaded %d files to Google Cloud Storage", len(s3_objects))
+
+    def transfer_files_async(self, files: list[str], gcs_hook: GCSHook, s3_hook: S3Hook):
+        """Submits Google Cloud Storage Transfer Service job to copy files from AWS S3 to GCS."""
+        if len(files) <= 0:
```

Comment:
Should we just go with an empty check?

```suggestion
        if not len(files):
```